### ISSUE TITLE
Replace `github.com/sirupsen/logrus`

### ISSUE THREAD

**Issue by na--** [grafana/k6#2958]
[github.com/sirupsen/logrus](https://github.com/sirupsen/logrus) is the structured logging library that has been used in k6 since literally the second commit in the repo, https://github.com/grafana/k6/commit/9d2b96c89b0f425112962325213419764e3f8baa :exploding_head: 

However, it has its fair share of API issues that we have had to work around, it doesn't have the greatest performance or feature set, and it has been in maintenance mode for the last 3+ years (https://github.com/sirupsen/logrus/commit/4ecd9a62bd57b6533e0d87ae23a6b10adc32b5ce). So it has been a low-key goal to replace it with something else for a while...

There are [a bunch of other structured logging Go libraries](https://github.com/avelino/awesome-go#logging) out there that are better in various ways. Previous discussions and issues like https://github.com/grafana/k6/issues/216 advocated for switching to one of them. However, with the proposal from https://github.com/golang/go/issues/56345, it seems likely that Go will have structured logging support in the standard library at some future point :tada: I haven't looked at the proposal in detail, but it seemed reasonable at a very quick glance and it already has an implementation in https://pkg.go.dev/golang.org/x/exp/slog that people can play around with! So, barring any show-stopping issues, I'd probably prefer to switch k6 to use it instead of another external dependency...

Regardless of which library we choose to go with, the initial steps we need to do seem fairly clear to me. I think we should first do a few small refactors of the k6 code and get rid of any non-interface logrus types we rely one, especially the [`*logrus.Logger`](https://pkg.go.dev/github.com/sirupsen/logrus#Logger) variables we have in a few central places like these: 
https://github.com/grafana/k6/blob/0b1fe61c5762c83c52a2378f6cdf9c0c3ff1f53d/cmd/state/state.go#L53
https://github.com/grafana/k6/blob/0b1fe61c5762c83c52a2378f6cdf9c0c3ff1f53d/lib/test_state.go#L19-L20
https://github.com/grafana/k6/blob/0b1fe61c5762c83c52a2378f6cdf9c0c3ff1f53d/lib/vu_state.go#L38-L39

As the inline comments say, we should be able to easily replace all usage of the concrete `*logrus.Logger` type with the [`logrus.FieldLogger` interface](https://pkg.go.dev/github.com/sirupsen/logrus#FieldLogger) it implements, once the downstream APIs are also adjusted. Other concrete types like [`logrus.Entry`](https://pkg.go.dev/github.com/sirupsen/logrus#Entry) and [`logrus.Fields`](https://pkg.go.dev/github.com/sirupsen/logrus#Fields) might be a bit trickier to replace. At least some of the usages of `*logrus.Entry` should be easy to substitute :crossed_fingers: Because `logrus.Entry` also implements the `logrus.FieldLogger` interface, the result of any operation that returns a `*logrus.Entry` (e.g. `logger.WithFields()`), we should be able to assign to a `logrus.FieldLogger` variable. 

For an easier transition, maybe we should actually not use the `logrus.FieldLogger` type directly, but our own type that initially matches that logrus interface, but is then gradually evolved to our desired end state :thinking: For example, we probably want to prune some of the methods from that interface we don't use, instead of re-implementing them :thinking: 

In any case, after all of that prep work is done, we should be able to add a shim layer that implements the same interface ( `logrus.FieldLogger` or our own equivalent) that is now used everywhere, but uses our new logging library (e.g. [slog](https://pkg.go.dev/golang.org/x/exp/slog)) to actually log stuff. Hopefully, this would require only a small PR with some changes to the few places in `cmd/` where we initialize the loggers :crossed_fingers: Everything else underneath that actually uses the logger should be left for later, changed one component at a time in separate PRs :crossed_fingers: :pray: 

After we have moved everything to the new library, we can get rid of the logrus dependency :tada: But we'd have basically stopped using it as soon as we start using the new library in `cmd/`, the rest would be cleanup of its types.

**Comment by na--** [comment]
Hmm, looking into this a bit more, I discovered it's not as simple as I'd initially hoped :disappointed: 

It seems like we _must_ have our own `interface` type during the transition, since the [`logrus.FieldLogger` interface](https://pkg.go.dev/github.com/sirupsen/logrus#FieldLogger) doesn't actually have the `AddHook` method of [`*logrus.Logger`](https://pkg.go.dev/github.com/sirupsen/logrus#Logger), and we depend on it in a bunch of places (mostly tests, but also for the [`--log-output=loki`](https://k6.io/docs/using-k6/k6-options/reference/#log-output) integration in [log/loki.go](https://github.com/grafana/k6/blob/master/log/loki.go))

**Comment by inancgumus** [comment]
Yes, it's definitely a bit more. [k6-browser also needs more methods](https://github.com/grafana/xk6-browser/issues/818#issuecomment-1460081431) that we might want to put into the new interface(s). I'm thinking of a better way for the k6-browser's case 🤔

**Comment by MrMegaMango** [comment]
We are on zap.logger trying out grafana and don't want to migrate to logrus
